src/System.Private.CoreLib/shared/System/Text/DecoderBestFitFallback.cs

```diff
@@ -62,17 +63,17 @@ internal sealed class InternalDecoderBestFitFallbackBuffer : DecoderFallbackBuff
         private InternalDecoderBestFitFallback _oFallback;
 
         // Private object for locking instead of locking on a public type for SQL reliability work.
-        private static object s_InternalSyncObject;
+        private static object? s_InternalSyncObject;
         private static object InternalSyncObject
         {
             get
             {
                 if (s_InternalSyncObject == null)
                 {
                     object o = new object();
-                    Interlocked.CompareExchange<object>(ref s_InternalSyncObject, o, null);
+                    Interlocked.CompareExchange<object?>(ref s_InternalSyncObject, o, null);
                 }
-                return s_InternalSyncObject;
+                return s_InternalSyncObject!;
```

Comment:
This `!` shouldn't be necessary once we attribute CompareExchange.  Can you please add a `// TODO-NULLABLE: https://github.com/dotnet/roslyn/issues/26761` comment here (and in any other similar places) so we can track these down more easily in the future?